### PR TITLE
Daily prorated leave recovered changes

### PR DESCRIPTION
**Pull request by yusuffGoodie**
## Is this a Feature, Chore or Bug?
- [*] Feature
- [] Chore
- [] Bug

## Analysis and design (optional)
Analyse and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.

## Is there a business logic within a doctype?
    - [*] Yes
    - [] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.

## Areas affected and ensured
List out the areas affected by your code changes.

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.

## Did you test with the following dataset?
- [*] Existing Data
- [] New Data

## Was child table created?
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [] Yes
    - [*] No
        If yes, did you write a delete patch?

## Is patch required?
- [] Yes
- [*] No
    ## Was the patch test?

## Which browser(s) did you use for testing?
  - [*] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/utils.py [hunk 1]
```diff
@@ -961,10 +975,8 @@ def get_new_leave_allocated_for_annual_paid_leave(allocation, leave_type):
     new_leaves_allocated = 0
 
     # Fetch employee annual leave allocation from employee
-    employee_annual_leave = frappe.db.get_value('Employee', allocation.employee, 'annual_leave_balance')
```
Comment (NoBoneZ) [inline one_fm/utils.py:964]: Please you removed a block of code here, is this intentional ?

File: one_fm/utils.py [hunk 2]
```diff
@@ -898,38 +898,52 @@ def increase_daily_leave_balance():
     '''
 
     # Get List of Leave Allocation for today of a Leave Type (having Is Paid Annual Leave marekd True)
-    allocation_list = get_paid_annual_leave_allocation_list()
```
Comment (NoBoneZ) [inline one_fm/utils.py:901]: Please you removed a block here, is this intentional ?

File: one_fm/utils.py (hunk 1 above)
Comment (yusuffGoodie) [reply one_fm/utils.py:964]: Yes

File: one_fm/utils.py (hunk 2 above)
Comment (yusuffGoodie) [reply one_fm/utils.py:901]: Yes

File: one_fm/utils.py (hunk 1 above)
Comment (yusuffGoodie) [reply one_fm/utils.py:964]: I don't remove it. I just wrap it in try and catch

Comment (NoBoneZ) [reply one_fm/utils.py:964]: please, if you observe it below, the block was removed, and replaced with a single query

**Review by NoBoneZ — APPROVED** [review]

**Pull request merged**